internal/backend/connector_updates.go

```diff
@@ -503,78 +504,104 @@ func (user *user) applyMessageDeleted(ctx context.Context, update *imap.MessageD
 func (user *user) applyMessageUpdated(ctx context.Context, update *imap.MessageUpdated) error {
 	var stateUpdates []state.Update
 
-	if err := user.db.Write(ctx, func(ctx context.Context, tx *ent.Tx) error {
-		var exists bool
+	log := logrus.WithField("message updated", update.Message.ID.ShortID())
 
+	if err := user.db.Write(ctx, func(ctx context.Context, tx *ent.Tx) error {
 		internalMessageID, err := db.GetMessageIDFromRemoteID(ctx, tx.Client(), update.Message.ID)
 		if err != nil {
-			if !ent.IsNotFound(err) {
-				return err
+			if ent.IsNotFound(err) {
+				log.Debugf("Message %v no longer exists, skipping", update.Message.ID.ShortID())
+				return nil
 			}
-		} else {
-			exists = true
+			return err
 		}
 
-		// delete the message and remove from the mailboxes.
-		if exists {
-			mailboxes, err := db.GetMessageMailboxIDs(ctx, tx.Client(), internalMessageID)
-			if err != nil {
-				return err
+		// compare and see if the literal has changed.
+		onDiskLiteral, err := user.store.Get(internalMessageID)
+		if err != nil {
+			return fmt.Errorf("failed to retrieve literal from cache")
```

Comment:
```suggestion
			return fmt.Errorf("failed to retrieve literal from cache: %w", err)
```

might be useful to show the underlying error

Reply:
Done